src/Widgets/BridgeWidgetApi.ts

```diff
@@ -156,7 +156,7 @@ export class BridgeWidgetApi {
         if (!connection.provisionerUpdateConfig || !connection.getProvisionerDetails)  {
             throw new ApiError("Connection type does not support updates", ErrCode.UnsupportedOperation);
         }
-        connection.provisionerUpdateConfig(req.userId, req.body);
+        await connection.provisionerUpdateConfig(req.userId, req.body);
```

Comment:
@Half-Shot You were correct to have [questioned](https://github.com/matrix-org/matrix-hookshot/pull/505/commits/08c70038d1ce14af1701b8ac01bec0b2823b5c25#diff-2aa3eec27e7c4d9bb2c3598bdcf50971052c917cb826c58bb00884430d1d3aa8) the promise-ness of this :slightly_smiling_face:   Turns out the real culprit was an improper return type on the method declaration (`void` instead of `Promise<void>`).